server.py

```diff
@@ -1,42 +1,57 @@
-import socket
+import socket as s
 
-ADDR = ('127.0.0.1', 8000)
-socket = socket.socket(
-    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+
+def response_ok(reason_phrase):
+    return "HTTP/1.1 200" + reason_phrase + "\r\n"
+    "Content-Type: text/html; charset=utf-8\r\n"
+
+
+def response_error(error_code, reason_phrase):
+    response = "HTTP/1.1 " + error_code + " " + reason_phrase + "\r\n"
+    "Content-Type: text/html; charset=utf-8\r\n"
+    return response
+
+
+def return_request():
+    ADDR = ('127.0.0.1', 8000)
+    socket = s.socket(
+        s.AF_INET, s.SOCK_STREAM, s.IPPROTO_IP
     )
-socket.bind(ADDR)
-socket.listen(1)
-
-
-def response_ok():
-    return conn.sendall(
-        "HTTP/1.1 200 OK\n"
-        "Content-Type: text/html; charset=utf-8\n"
-        "<!DOCTYPE html>\n"
-        "<html>\n"
-        "\t<head>\n"
-        "\t\t<title>Status 200 OK</title>\n"
-        "\t</head>\n"
-        "\t<body>\n"
-        "\t\t<p>HTTP:200 OK response</p>\n"
-        "\t</body>\n"
-        "</html>\n"
-        )
-
-
-while True:
-    try:
-        conn, addr = socket.accept()
-        output = ""
-        while True:
-            msg = conn.recv(16)
-            output = output + msg
-            if len(msg) < 16:
-                response_ok()
-                conn.sendall(output)
-                conn.close()
-                break
-    except KeyboardInterrupt:
-        break
-
-# some change
+
+    socket.bind(ADDR)
+    socket.listen(1)
+    while True:
+        try:
+            conn, addr = socket.accept()
+            output = ""
+            while True:
+                msg = conn.recv(16)
+                output = output + msg
+                if len(msg) < 16:
+                    break
+            try:
+                final = parse_request(output)
+                conn.sendall(response_ok(final))
+            except ValueError as detail:
+                detail = str(detail)
+                conn.sendall(response_error("400", detail))
+            conn.close()
+        except KeyboardInterrupt:
+            break
+
+
+def parse_request(request):
+    if "GET" not in request:
+        raise ValueError("Error.  Must send GET request")
+    if "HTTP/1.1" not in request:
+        raise ValueError("Error. Only accepts HTTP 1.1 Protocol")
+    if "Host:" not in request:
+        raise ValueError("Error. Must provide host")
```

Comment:
while each of these statements needs to be true in order to have a good request, just having them be true is not enough to prove that the request is good.  Consider this HTTP request:

```
GETMEASAMMICH / HTTP/1.0<CRLF>
OtherHeader: HTTP/1.1<CRLF>
BadHeader: Host:foobar<CRLF>
<CRLF>
```

This will pass all your tests, but still not be any good.  It's not easy to do this job absolutely correctly, but you can do better than this